lib/protocol/bep_extensions.go

```diff
@@ -266,24 +266,37 @@ func BlocksEqual(a, b []BlockInfo) bool {
 }
 
 func (f *FileInfo) SetMustRescan(by ShortID) {
-	f.LocalFlags = FlagLocalMustRescan
-	f.ModifiedBy = by
-	f.Blocks = nil
-	f.Sequence = 0
+	f.setLocalFlags(by, FlagLocalMustRescan)
 }
 
 func (f *FileInfo) SetIgnored(by ShortID) {
-	f.LocalFlags = FlagLocalIgnored
-	f.ModifiedBy = by
-	f.Blocks = nil
-	f.Sequence = 0
+	f.setLocalFlags(by, FlagLocalIgnored)
 }
 
 func (f *FileInfo) SetUnsupported(by ShortID) {
-	f.LocalFlags = FlagLocalUnsupported
+	f.setLocalFlags(by, FlagLocalUnsupported)
+}
+
+func (f *FileInfo) SetDeleted(by ShortID) {
+	f.ModifiedBy = by
+	f.Deleted = true
+	f.Version = f.Version.Update(by)
+	f.ModifiedS = time.Now().Unix()
+	f.setNoContent()
+}
+
+func (f *FileInfo) setLocalFlags(by ShortID, flags uint32) {
+	f.RawInvalid = false
+	f.LocalFlags = flags
 	f.ModifiedBy = by
-	f.Blocks = nil
 	f.Sequence = 0
```

Comment:
Not clear to me why this one sets sequence to zero but not the others...

Reply:
Me neither, didn't want to change existing behaviour though. I don't think it matters, as for local it gets a new sequence number when adding to db and for remotes sequence isn't used (and I don't see any reason to modify a fileinfo for a remote before adding to db). However I also don't see any point in resetting it.

I'll prune it, call out if you can think of any reason to keep it or just think "better don't touch it".